src/sync/impl/fetchLocal.js

```diff
@@ -11,45 +11,77 @@ import {
 import allPromisesObj from '../../utils/fp/allPromisesObj'
 import type { Database, Collection, Model } from '../..'
 import * as Q from '../../QueryDescription'
+import type { Clause } from '../../QueryDescription'
 import { columnName } from '../../Schema'
 
 import type { SyncTableChangeSet, SyncLocalChanges } from '../index'
 
 // NOTE: Two separate queries are faster than notEq(synced) on LokiJS
 const createdQuery = Q.where(columnName('_status'), 'created')
 const updatedQuery = Q.where(columnName('_status'), 'updated')
+const deletedQuery = Q.where(columnName('_status'), 'deleted')
 
-async function fetchLocalChangesForCollection<T: Model>(
+type fetchLocalChangesForCollectionArgs<T> = {
   collection: Collection<T>,
-): Promise<[SyncTableChangeSet, T[]]> {
+  createdQueries?: Clause[],
+  updatedQueries?: Clause[],
+  deletedQueries?: Clause[],
+}
+
+async function fetchLocalChangesForCollection<T: Model>({
+  collection,
+  createdQueries = [],
+  updatedQueries = [],
+  deletedQueries = [],
+}: fetchLocalChangesForCollectionArgs<T>): Promise<[SyncTableChangeSet, T[]]> {
   const [createdRecords, updatedRecords, deletedRecords] = await Promise.all([
-    collection.query(createdQuery).fetch(),
-    collection.query(updatedQuery).fetch(),
-    collection.database.adapter.getDeletedRecords(collection.table),
```

Comment:
Yes, I totally agree with you on this point and that's why I mentioned `It required more changes than I expected and I'm not sure I'm happy with it.`. 

At the same time, I think we should have options to query deleted especially to avoid `collection.database.adapter.getDeletedRecords(collection.table)`, that basically breaks the `model => collection => database` structure (part of the decoupling I mentioned).

Maybe changing to `collection.query(updatedQuery).fetchIds()` to be a very specific method.

I'm not happy with this solution, especially on the part that skips the cache, but I think there are too many hard-coded rules (`createdQuery`, `updatedQuery`) that make sense to be customizable IMHO.

In short, I'm not happy with the purposed solution, but I think it's worth thinking about the concept of customization.

Even the `draft` state might be troublesome, as some users may want the `drafts` to be synced. An email app would like `drafts` to be synced. And in this case, they would need another `status` field.

I believe that #1027 starts a discussion about `pullable, pushable, local-only, read-only` scenarios.